src/extensions/mvs/additions/annotation-prop.ts

```diff
@@ -0,0 +1,380 @@
+        const annots = await Annotations.fromSpecs(ctx, specs, data);
+        return { value: annots } satisfies CustomProperty.Data<Annotations>;
+    }
+});
+
+
+/** Represents multiple annotations retrievable by their ID */
+export class Annotations {
+    private constructor(private dict: { [id: string]: Annotation }) { }
+    static async fromSpecs(ctx: CustomProperty.Context, specs: AnnotationSpec[], model?: Model): Promise<Annotations> {
+        const sources: AnnotationSource[] = specs.map(annotationSourceFromSpec);
+        const files = await getFilesFromSources(ctx, sources, model);
+        const annots: { [id: string]: Annotation } = {};
+        for (let i = 0; i < specs.length; i++) {
+            const spec = specs[i];
+            try {
+                const file = files[i];
+                if (!file.ok) throw file.error;
+                annots[spec.id] = await Annotation.fromSpec(ctx, spec, file.value);
+            } catch (err) {
+                console.error(`Failed to obtain annotation (${err}).\nAnnotation specification:`, spec);
+                annots[spec.id] = Annotation.createEmpty(spec.schema);
+            }
+        }
+        return new Annotations(annots);
+    }
+    getAnnotation(id: string): Annotation | undefined {
+        return this.dict[id];
+    }
+    getAllAnnotations(): Annotation[] {
+        return Object.values(this.dict);
+    }
+}
+
+
+/** Retrieve annotation with given `annotationId` from custom model property "Annotations" and the model from which it comes */
+export function getAnnotationForStructure(structure: Structure, annotationId: string): { annotation: Annotation, model: Model } | { annotation: undefined, model: undefined } {
+    const models = structure.isEmpty ? [] : structure.models;
+    for (const model of models) {
+        if (model.customProperties.has(AnnotationsProvider.descriptor)) {
+            const annots = AnnotationsProvider.get(model).value;
+            const annotation = annots?.getAnnotation(annotationId);
+            if (annotation) {
+                return { annotation, model };
+            }
+        }
+    }
+    return { annotation: undefined, model: undefined };
+}
+
+/** Main class for processing annotation */
+export class Annotation {
+    /** Store mapping `ElementIndex` -> annotation row index for each `Model`, -1 means no row applies */
+    private indexedModels = new Map<UUID, number[]>();
+    private rows: AnnotationRow[] | undefined = undefined;
+
+    constructor(
+        public data: AnnotationData,
+        public schema: AnnotationSchema,
+    ) { }
+
+    /** Create a new `Annotation` based on specification `spec`. Use `file` if provided, otherwise download the file.
+     * Throw error if download fails or problem with data. */
+    static async fromSpec(ctx: CustomProperty.Context, spec: AnnotationSpec, file?: AnnotationFile): Promise<Annotation> {
+        file ??= await getFileFromSource(ctx, annotationSourceFromSpec(spec));
```

Comment:
Cool, didn't know this operator existed.

Reply:
My favorite :D 
And you can do this with it:
```
let cachedStuff = undefined;
function getStuff(){ return cachedStuff ??= actuallyComputeStuff(); }
```